src/solver/optimisation/opt_rename_problem.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+#include "opt_structure_probleme_a_resoudre.h"
+#include "opt_export_structure.h"
+
+const std::string SEPARATOR = "::";
+const std::string AREA_SEP = "$$";
+
+std::string BuildName(const std::string& name,
+                      const std::string& location,
+                      const std::string& timeIdentifier);
+
+class TargetVectorUpdater
+{
+public:
+    TargetVectorUpdater(bool isRenamingProcessed, std::vector<std::string>& target) :
+     target_(target)
+    {
+        if (isRenamingProcessed)
+        {
+            UpdateTargetAtIndex = std::bind(
+              &TargetVectorUpdater::UpdateTargetAtIndexImpl, this, std::placeholders::_1, std::placeholders::_2);
+        }
+    }
+
+    std::function<void(const std::string&, unsigned index)> UpdateTargetAtIndex
+      = [](const std::string&, unsigned index) {};
+
+    std::string& StringAtIndex(unsigned index) const
+    {
+        return target_[index];
+    }
+
+private:
+    std::vector<std::string>& target_;
+    void UpdateTargetAtIndexImpl(const std::string& full_name, unsigned index)
+    {
+        target_[index] = full_name;
+    }
+};
+
+struct CurrentAssetsStorage
```

Comment:
class **CurrentAssetsStorage** reveals to be cumbersome ! 
We should move its content to class **Namer** : 
```C++
class Namer 
{
...
   void UpdateTimeStep(int timeStep)
    {
        current_timeStep_  = timeStep;
    }

    void UpdateArea(const std::string& area)
    {
        current_area_ = area;
    }
protected :        
     int current_timeStep_ = 0;
     std::string current_origin_;
     std::string current_destination_;
     std::string current_area_;
 ...
}
```